dashboard/app/models/sections/omni_auth_section.rb

```diff
@@ -27,18 +27,33 @@
 
 class OmniAuthSection < Section
   def self.from_omniauth(code:, type:, owner_id:, students:)
-    oauth_section = where(code: code).first_or_create! do |section|
+    oauth_section = with_deleted.where(code: code).first_or_create! do |section|
       section.name = 'New Section'
       section.user_id = owner_id
       section.login_type = type
     end
-    students.each do |student|
-      oauth_section.add_student User.from_omniauth(student, {'user_type' => User::TYPE_STUDENT})
+
+    oauth_section.restore if oauth_section.deleted?
```

Comment:
What does the `recovery_window` do?  The spec is for users to be able to re-import an OAuth classroom section, even if they've previously deleted one with the same code.